File: 08_SoundTools/08_SoundTools/ViewController.m
//
//  ViewController.m
//  08_SoundTools
//
//  Created by GeekerHua on 15/12/30.
//  Copyright © 2015年 GeekerHua. All rights reserved.
//

#import "ViewController.h"
//#import "08_SoundTools-Bridging-Header.h"
#import "单例-Swift.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
  
//    [SoundTools sharedTools]
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event{
    for (int i = 0; i < 2; i++) {
        NSLog(@"%@",[SoundTools sharedTools]);
        
    }
    NSLog(@"----%@",[[SoundTools alloc] init]);

}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: 05-getset/05-getset/ViewController.swift
//
//  ViewController.swift
//  05-getset
//
//  Created by GeekerHua on 15/12/30.
//  Copyright © 2015年 GeekerHua. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    // 懒加载 ,实际上是一个闭包
    lazy var dataList: [String] = {
        print("我懒了")
        return ["张三", "李四"]
    }()
    
    // 简单写法
    lazy var list:[String] = ["zhagnsan", "lisi"]
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        let p = Person()
        p.model = "哈哈"
//        print(dataList)
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func touchesBegan(touches: Set<UITouch>, withEvent event: UIEvent?) {
        print(dataList)
    }


}

